### ISSUE TITLE
Missing libraries in the Docker image

### ISSUE THREAD

**Issue by Ambros94** [ElementsProject/lightning#3074]
I tried to build a docker image of the repository locally on my mac but at runtime it looks like some libraries are missing (or have not linked properly during building).

Namely: 
1. libgmp.so.10
2. sqlite

## Local environment information
macOS 10.14.16 (18G95)
Docker version 19.03.1, build 74b1e89

Steps to 
```bash
git clone https://github.com/ElementsProject/lightning.git
cd lightning
docker build -t lightning .
docker run -ti lightning
```
### Output

```
lightningd: error while loading shared libraries: libgmp.so.10: cannot open shared object file: No such file or directory
```

**Comment by NicolasDorier** [comment]
I don't think it should be closed, I think https://github.com/ElementsProject/lightning/pull/3101 does not fix it.

**Comment by NicolasDorier** [comment]
it is quite strange that you can't build, I can.

**Comment by NicolasDorier** [comment]
Did something changed on master that require dynamic lib? The best is normally to build c-lightning statically.

**Comment by NicolasDorier** [comment]
Also the fix in #3101 does not fix the arm32 and arm64 build. Has there been a breaking change in the build system that does not allow to statically link the dependencies?

**Comment by NicolasDorier** [comment]
Second problem: It is possible, because of docker cache that the version of lib in the build image is not the same as the runtime image (this is why I want to statically link). This issue already happened to me.
